lib/ansible/executor/task_executor.py

```diff
@@ -526,6 +526,10 @@ def _execute(self, variables=None):
             display.debug("running the handler")
             try:
                 result = self._handler.run(task_vars=variables)
+            except AnsibleActionSkip as e:
+                return dict(skipped=True, msg=to_text(e))
+            except AnsibleActionFail as e:
+                return dict(failed=True, msg=to_text(e))
```

Comment:
Need to make use of e.obj here if we're using that for additional result information.  something like:

``` python
results = {}
if self._obj is not None:
    results = self._obj.copy()
results['failed'] = True
results['msg'] = to_text(e))
return results
```

Might want to move some or all of that into the exception class instead of having it here, when processing the results of the action plugin.run().